### PR TITLE
AixPB: add zlib_links role

### PR DESCRIPTION
**Pull request by Haroon-Khel**
Ref https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1952

@aixtools found that creating the following symlinks helps to resolve a zlib issue mentioned in the referenced issue
```
/usr/include/zconf.h -> /opt/freeware/include/zconf.h
/usr/include/zlib.h -> /opt/freeware/include/zlib.h
```
The symlinks were created on test-ibm-aix71-ppc64-1, which allowed them to pass a system zlib check which was previously failing

### CODE REVIEW

**Review by sxa — CHANGES_REQUESTED** [review]
I'd prefer to do this within the `yum` role since it's dependent on the `zlib` package that comes from there and we also set up perl symlinks within that role.

**Review by sxa — APPROVED** [review]
I'll "tentatively" approve this (since at the moment it's broken) although I do worry about whether there is a possibility of another zlib having been installed on the machine in the default system locations which this would overwrite.

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/yum/tasks/main.yml [hunk 1]
```diff
@@ -104,3 +104,33 @@
   tags:
     - rpm_install
     - yum
+
+# Create zlib.h and zconf.h links
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/yum/tasks/main.yml:108]: We don't say why...

**Review by karianna — CHANGES_REQUESTED** [review]

**Comment by Haroon-Khel** [comment]
@karianna Ive included a reference to the infra issue in the comment

Ive also tested this pr on an aix test environment. It works as expected 👍🏻

**Review by karianna — APPROVED** [review]

**Comment by aixtools** [comment]
Actually researching these things is something I do. You just need to ask.

Did anyone notice that I knew what the issue was before even looking? I have had a lot of experience with getting AIX systems to build using the libraries provided by AIX bos (in this case rpm.rte).

There are other ways to resolve this, e.g., use a playbook to copy https://github.com/madler/zlib/blob/master/zconf.h and https://github.com/madler/zlib/blob/master/zlib.h to /usr/include - without all the overhead of yum and all the other dependencies this brings.

And when the IBM provided libz.a is not sufficient - open a ticket with IBM to fix that. Having a core library installed twice (in the same location!) by two different package managers - is just asking for trouble - and you found it - again.

This is why - somewhere between 2008 and 2010 I stopped using the RPM prepared packages and started building everything on my own - and as I got better at packaging - created `aixtools`.

But, you need to want to really know what the demands of OpenJava are - and use only that - rather than install stuff blindly.

Again, I am willing to do the research and documentation. But someone needs to ask - so I know someone cares and is prepared to listen when I finish.

** Guess I am too late - just noticed this is tagged as `done` 5 hours ago. Oh well.

**Comment by aixtools** [comment]
This is the zlib setup (for the library, not the include file) on test-osuosl-aix72-ppc64-1 (aka ojdk03):
```
# rpm -ql zlib
/opt/freeware/doc/zlib-1.2.11
/opt/freeware/doc/zlib-1.2.11/ChangeLog
/opt/freeware/doc/zlib-1.2.11/FAQ
/opt/freeware/doc/zlib-1.2.11/README
/opt/freeware/lib/libz.a
# ar tv /opt/freeware/lib/libz.a
rwxr-xr-x   205/1     164939 Dec 17 09:29 2020 libz.so.1
rwxr-xr-x   205/1     164939 Dec 17 09:29 2020 shr.o
# ar tv /usr/opt/rpm/lib/libz.a
rwxr-xr-x   211/1     149613 Feb 14 03:02 2017 libz.so.1
rwxr-xr-x   211/1       8350 Feb 14 03:02 2017 shr.o
# ls -l /usr/lib/libz.a
lrwxrwxrwx    1 root     system           23 Feb 11 09:41 /usr/lib/libz.a -> /usr/opt/rpm/lib/libz.a
# lslpp -w /usr/opt/rpm/lib/libz.a
  File                                        Fileset               Type
  ----------------------------------------------------------------------------
  /usr/opt/rpm/lib/libz.a                     rpm.rte               File
```

The same commands on build-osuosl-aix71-ppc64-1 (aka ojdk01):
```
# ls -l /usr/lib/libz.a
lrwxrwxrwx 1 root system 23 Feb 17 06:41 /usr/lib/libz.a -> /usr/opt/rpm/lib/libz.a
# rpm -ql zlib
/opt/freeware/doc/zlib-1.2.11
/opt/freeware/doc/zlib-1.2.11/ChangeLog
/opt/freeware/doc/zlib-1.2.11/FAQ
/opt/freeware/doc/zlib-1.2.11/README
/opt/freeware/lib/libz.a
# rpm -q zlib
zlib-1.2.11-2.ppc
# lslpp -w /usr/opt/rpm/lib/libz.a
  File                                        Fileset               Type
  ----------------------------------------------------------------------------
  /usr/opt/rpm/lib/libz.a                     rpm.rte               File
# ar tv /opt/freeware/lib/libz.a
rwxr-xr-x   205/1     164939 Dec 17 07:29 2020 libz.so.1
rwxr-xr-x   205/1     164939 Dec 17 07:29 2020 shr.o
# ar tv /usr/opt/rpm/lib/libz.a
rwxr-xr-x   211/1     149613 Feb 14 01:02 2017 libz.so.1
rwxr-xr-x   211/1       8350 Feb 14 01:02 2017 shr.o
```

So, it appears that on these two systems the actual run-time library being referenced SHOULD be `/usr/opt/rpm/lib/libz.a`, but when /opt/freeware/lib is pre-pended to LIBPATH (which I believe I read in another PR) - a setup like that defeats expected AIX run-time support.

Notably - none of this has anything to do with zlib.h and/or zconf.h (that ./configure does not look for). But to get those files - zlib-devel needs to be installed - and it expects the rpm version of zlib to be installed.

* This is just one example of how good intentions have the potential to make AIX platform less stable.

* small differences that don't always have impact (for libz.so.1 doubtful that it needs something from /usr/vac/lib - but the AIXToolbox packages don't look there at all)

```
# dump -H /usr/opt/rpm/lib/libz.a

/usr/opt/rpm/lib/libz.a[libz.so.1]:

                        ***Loader Section***
                      Loader Header Information
VERSION#         #SYMtableENT     #RELOCent        LENidSTR
0x00000001       0x00000074       0x00000120       0x0000002b

#IMPfilID        OFFidSTR         LENstrTBL        OFFstrTBL
0x00000002       0x00001880       0x00000411       0x000018ab


                        ***Import File Strings***
INDEX  PATH                          BASE                MEMBER
0      /usr/vac/lib:/usr/lib:/lib
1                                    libc.a              shr.o

/usr/opt/rpm/lib/libz.a[shr.o]:

                        ***Loader Section***
                      Loader Header Information
VERSION#         #SYMtableENT     #RELOCent        LENidSTR
0x00000001       0x00000065       0x00000000       0x00000022

#IMPfilID        OFFidSTR         LENstrTBL        OFFstrTBL
0x00000002       0x00000998       0x000003f8       0x000009ba


                        ***Import File Strings***
INDEX  PATH                          BASE                MEMBER
0      /usr/lib:/lib
1                                    libz.a              libz.so.1
# dump -H /opt/freeware/lib/libz.a

/opt/freeware/lib/libz.a[libz.so.1]:

                        ***Loader Section***
                      Loader Header Information
VERSION#         #SYMtableENT     #RELOCent        LENidSTR
0x00000001       0x00000074       0x00000120       0x0000001e

#IMPfilID        OFFidSTR         LENstrTBL        OFFstrTBL
0x00000002       0x00001880       0x00000411       0x0000189e


                        ***Import File Strings***
INDEX  PATH                          BASE                MEMBER
0      /usr/lib:/lib
1                                    libc.a              shr.o

/opt/freeware/lib/libz.a[shr.o]:

                        ***Loader Section***
                      Loader Header Information
VERSION#         #SYMtableENT     #RELOCent        LENidSTR
0x00000001       0x00000074       0x00000120       0x0000001e

#IMPfilID        OFFidSTR         LENstrTBL        OFFstrTBL
0x00000002       0x00001880       0x00000411       0x0000189e


                        ***Import File Strings***
INDEX  PATH                          BASE                MEMBER
0      /usr/lib:/lib
1                                    libc.a              shr.o
```

As the saying goes - the devil is in the details.

**Pull request merged**